tools/update-snapcraft

```diff
@@ -0,0 +1,94 @@
+    return parser.parse_args()
+
+def get_latest_revision_from_release_file(path):
+    """Get the latest revision from yaml formatted release files."""
+    content = yaml.safe_load(path.read_text())
+    revisions = [release['version'] for release in content['releases']]
+    return f'{path.parent.stem}-last' \
+           if f'{path.parent.stem}-last' in revisions \
+           else sorted(revisions, key=cmp_to_key(semver.compare), reverse=True)[0]
+
+def get_latest_tempest_revision(release):
+    """Return the requirements entry for the latest tempest revision."""
+    return get_latest_revision_from_release_file(
+        Path(RELEASES_REPO_DIR) / 'deliverables' / release / 'tempest.yaml'
+    )
+
+def get_latest_plugin_requirements(release):
+    """Return list of requirements entries for the latest tempest plugin revisions."""
+    result = []
+    plugin_release_file_paths = [
+        entry
+        for entry in (Path(RELEASES_REPO_DIR) / 'deliverables' / release).iterdir()
+        if entry.is_file() and yaml.safe_load(entry.read_text())['type'] == 'tempest-plugin'
+    ]
+    for path in plugin_release_file_paths:
+        try:
+            latest_revision = get_latest_revision_from_release_file(path)
+            result.append(
+            OPENSTACK_REPO_URL_FMT.format(project=path.stem, ref=latest_revision)
+        )
+        except Exception as exception:
+            logger.warning(f'Skipping path:[{path}], error:[{repr(exception)}]')
+    return sorted(result)
+
+def get_latest_tempestconf_requirements():
+    """Return the requirements entry for the latest python-tempestconf revision."""
+    project = 'python-tempestconf'
+    latest_revision = feedparser.parse(PYPI_RSS_FEED_FMT.format(project=project)).entries[0].title
+    return OPENINFRA_REPO_URL_FMT.format(project=project, ref=latest_revision)
+
+def main(args):
+    """Entry point to the application."""
+    pygit2.clone_repository(RELEASE_REPO_URL, RELEASES_REPO_DIR)
+    snapcraft_yaml_path = Path(__file__).parent.parent / 'snap' / 'snapcraft.yaml'
+    snapcraft_yaml = yaml.safe_load(snapcraft_yaml_path.read_text())
+
+    snapcraft_yaml['parts']['tempest']['source-tag'] = get_latest_tempest_revision(args.release)
+    snapcraft_yaml['parts']['tempest']['python-packages'] = ['confluent-kafka==1.8.2']
+    snapcraft_yaml['parts']['tempest']['python-packages'].extend(get_latest_plugin_requirements(args.release))
+    snapcraft_yaml['parts']['tempest']['python-packages'].append(get_latest_tempestconf_requirements())
+
+    snapcraft_yaml_path.write_text(
+        yaml.safe_dump(snapcraft_yaml, sort_keys=False)
+    )
+    return 0
+
+def str_presenter(dumper, data):
+    """Configure yaml for dumping multiline strings."""
+    if len(data.splitlines()) > 1:
+        return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='|')
+    return dumper.represent_scalar('tag:yaml.org,2002:str', data)
+
+if __name__ == '__main__':
+    import sys
```

Comment:
why import here?

Reply:
since `sys` was only needed if the file is run, not if it is imported, i had done it that way. added to the top of the file in the update.